### PR TITLE
Bug 1340787 - Maybe fix runnable_jobs api for comm-* trees

### PR DESCRIPTION
**Pull request by KWierso**
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/2191)

### CODE REVIEW

File: treeherder/etl/runnable_jobs.py [hunk 1]
```diff
@@ -123,6 +123,9 @@ def _taskcluster_runnable_jobs(project, decision_task_id):
     tc_graph = {}
     if not decision_task_id:
         decision_task_id = _query_latest_gecko_decision_task_id(project)
+    # If we still don't have task id, lets just bail out now
```
Comment (wlach) [inline treeherder/etl/runnable_jobs.py]: This is a reasonable thing to do, but the comment makes it look like a hack. I would reword this to something like:

```py
# Some trees (e.g. comm-central) don't have a decision task, which means there are no taskcluster runnable jobs
```

File: treeherder/etl/runnable_jobs.py [hunk 2]
```diff
@@ -218,7 +221,10 @@ def list_runnable_jobs(project, decision_task_id=None):
...
     logger.info('Fetching {}'.format(url))
-    latest_task = fetch_json(url)
-    task_id = latest_task['taskId']
-    logger.info('For {} we found the task id: {}'.format(project, task_id))
+    try:
+        latest_task = fetch_json(url)
+        task_id = latest_task['taskId']
+        logger.info('For {} we found the task id: {}'.format(project, task_id))
+    except:
+        task_id = ''
```
Comment (wlach) [inline treeherder/etl/runnable_jobs.py]: This kind of blanket try/except can mask errors. I would do something like:

```py
latest_task = fetch_json(url)
task_id = latest_task.get('taskId')
if task_id:
    logger.info('For {} we found the task id: {}'.format(project, task_id))
else:
    logger.info('Task id not found for {}'.format(project))

return task_id
```

Comment (KWierso) [inline treeherder/etl/runnable_jobs.py]: Doesn't help, as the exception is thrown in fetch_json (specifically in make_request from treeherder/etl/common.py)

**Comment by wlach** [comment]
Could you catch the specific exception then? The API should throw a 404
which I think we should be able to detect...

File: treeherder/etl/runnable_jobs.py [hunk 3]
```diff
@@ -218,7 +222,14 @@ def list_runnable_jobs(project, decision_task_id=None):
...
-    task_id = latest_task['taskId']
-    logger.info('For {} we found the task id: {}'.format(project, task_id))
+    try:
+        latest_task = fetch_json(url)
+        task_id = latest_task['taskId']
+        logger.info('For {} we found the task id: {}'.format(project, task_id))
+    except requests.exceptions.HTTPError as e:
+        # Specifically handle 404 errors, as it means there's no decision task on this push
+        if e.response.status_code == 404:
+            task_id = ''
```
Comment (edmorley) [inline treeherder/etl/runnable_jobs.py]: I think `None` might be best rather than the empty string?

**Comment by edmorley** [comment]
Could you also tweak the commit message to remove the "Attempt" part and make the first line not wrap (<=72 characters)? Perhaps something like:

```
Bug 1340787 - Fix runnable_jobs API for pushes with no decision task

This fixes its use for repositories such as comm-*.
```

File: treeherder/etl/runnable_jobs.py (hunk 3 above)
Comment (wlach) [inline treeherder/etl/runnable_jobs.py]: ++ to this.

I'd also like a log message at the info level, saying that we didn't find the decision task.
